eduHub/src/main/kotlin/com/arsiu/eduhub/controller/nats/assignment/AssignmentNatsControllerDeleteById.kt

```diff
@@ -18,19 +20,32 @@ class AssignmentNatsControllerDeleteById(
     override val subject: String = ASSIGNMENT_DELETE_BY_ID
     override val parser: Parser<DeleteByIdAssignmentRequest> = DeleteByIdAssignmentRequest.parser()
 
-    override fun handler(request: DeleteByIdAssignmentRequest): DeleteByIdAssignmentResponse =
-        runCatching {
-            val id = request.request.assignmentId.id
-            service.delete(id).block()
-            getSuccessResponse()
-        }.getOrElse { ex ->
-            getFailureResponse(ex.javaClass.simpleName, ex.toString())
+    override fun handler(request: DeleteByIdAssignmentRequest): Mono<DeleteByIdAssignmentResponse> {
+        val assignmentRequest = request.request
+
+        return when (assignmentRequest.requestCase) {
+            AssignmentRequest.RequestCase.ASSIGNMENT_ID ->
+                Mono.defer {
+                    val id = assignmentRequest.assignmentId.id
+                    service.delete(id).thenReturn(getSuccessResponse())
+                }.onErrorResume {
+                    Mono.just(getFailureResponse(it.javaClass.simpleName, it.toString()))
+                }
```

Comment:
extract this code to function, so when expression will be shorter

Reply:
`+`